stackrox/stackrox
# ROX-18826: watch jwt-key mount changes

## Description

The goal of this PR is to make Central listen to `jwt-key` updates in `central-tls` secret. This PR includes:
* listening to jwt-key modifications
* updating corresponding JWT token validator so that it uses actual value of public key
* updating corresponding token issuer so that it uses actual value of private key

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

Manually execute with port-forward located on port 8000 pointing to central:
1. Create API Token `old-token`
2. In terminal `export ROX_API_TOKEN=<old-token-value`
3. `roxctl central debug download-diagnostics --endpoint=localhost:8000` should be successful
4. export `JWT_KEY=<valid jwt key>`
5. `kubectl patch secret central-tls -n stackrox  --patch="{\"data\": { \"jwt-key.pem\": \"$JWT_KEY\" }}"`
6. `roxctl central debug download-diagnostics --endpoint=localhost:8000` should return an error
7. Create API Token `new-token`
8. In terminal `export ROX_API_TOKEN=<new-token-value`
9. `roxctl central debug download-diagnostics --endpoint=localhost:8000` should be successful
